Clover/app/src/main/java/org/floens/chan/utils/PostUtils.java

```diff
@@ -0,0 +1,135 @@
+package org.floens.chan.utils;
+
+import android.annotation.SuppressLint;
+
+import org.floens.chan.core.model.ChanThread;
+import org.floens.chan.core.model.Post;
+import org.floens.chan.core.model.orm.PostHide;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class PostUtils {
+
+    private PostUtils() {
+    }
+
+    public static Post findPostById(int id, ChanThread thread) {
```

Comment:
Maybe pass a list of posts here instead of ChanThread so we can get rid of ChanThread dependency here?

Reply:
Nah, it's more convenient this way.